neo/rawio/openephysbinaryrawio.py

```diff
@@ -0,0 +1,403 @@
+
+                # array annotations for signal channels
+                for stream_index, stream_name in enumerate(sig_stream_names):
+                    sig_ann = seg_ann['signals'][stream_index]
+                    d = self._sig_streams[0][0][stream_index]
+                    for k in ('identifier', 'history', 'source_processor_index',
+                              'recorded_processor_index'):
+                        if k in d['channels'][0]:
+                            values = np.array([chan_info[k] for chan_info in d['channels']])
+                            sig_ann['__array_annotations__'][k] = values
+
+                # array annotations for event channels
+                # use other possible datat in _possible_event_stream_names
+                for stream_index, stream_name in enumerate(event_stream_names):
+                    ev_ann = seg_ann['events'][stream_index]
+                    d = self._evt_streams[0][0][stream_index]
+                    for k in _possible_event_stream_names:
+                        if k in ('timestamps', ):
+                            continue
+                        if k in d:
+                            # split custom dtypes into separate annotations
+                            if d[k].dtype.names:
+                                for name in d[k].dtype.names:
+                                    ev_ann['__array_annotations__'][name] = d[k][name].flatten()
+                            else:
+                                ev_ann['__array_annotations__'][k] = d[k]
+
+    def _segment_t_start(self, block_index, seg_index):
+        return self._t_start_segments[block_index][seg_index]
+
+    def _segment_t_stop(self, block_index, seg_index):
+        return self._t_stop_segments[block_index][seg_index]
+
+    def _channels_to_group_id(self, channel_indexes):
+        if channel_indexes is None:
+            channel_indexes = slice(None)
+        channels = self.header['signal_channels']
+        group_ids = channels[channel_indexes]['group_id']
+        assert np.unique(group_ids).size == 1
+        group_id = group_ids[0]
+        return group_id
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        sigs = self._sig_streams[block_index][seg_index][stream_index]['memmap']
+        return sigs.shape[0]
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        t_start = self._sig_streams[block_index][seg_index][stream_index]['t_start']
+        return t_start
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        sigs = self._sig_streams[block_index][seg_index][stream_index]['memmap']
+        sigs = sigs[i_start:i_stop, :]
+        if channel_indexes is not None:
+            sigs = sigs[:, channel_indexes]
+        return sigs
+
+    def _spike_count(self, block_index, seg_index, unit_index):
+        pass
```

Comment:
Wouldn't it be safer to raise a `NotImplementedError` here in case someone starts extending the IO for handling spikes also?